src/Open3D/Visualization/Rendering/Filament/FilamentEntitiesMods.cpp

```diff
@@ -0,0 +1,227 @@
+    if (materialInstance_ != nullptr) {
+        utility::LogWarning(
+                "Previous material instance modifications are not finished!");
+    }
+
+    materialInstance_ = materialInstance;
+    currentHandle_ = id;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetParameter(const char* parameter,
+                                                         const int value) {
+    if (materialInstance_) {
+        materialInstance_->setParameter(parameter, value);
+    }
+
+    return *this;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetParameter(const char* parameter,
+                                                         const float value) {
+    if (materialInstance_) {
+        materialInstance_->setParameter(parameter, value);
+    }
+
+    return *this;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetParameter(
+        const char* parameter, const Eigen::Vector3f& v) {
+    if (materialInstance_) {
+        materialInstance_->setParameter(parameter,
+                                        math::float3{v(0), v(1), v(2)});
+    }
+
+    return *this;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetColor(
+        const char* parameter, const Eigen::Vector3f& value) {
+    if (materialInstance_) {
+        const auto color =
+                filament::math::float3{value.x(), value.y(), value.z()};
+        materialInstance_->setParameter(parameter, filament::RgbType::LINEAR,
+                                        color);
+    }
+
+    return *this;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetColor(
+        const char* parameter, const Eigen::Vector4f& value) {
+    if (materialInstance_) {
+        auto color =
+                filament::math::float4{value(0), value(1), value(2), value(3)};
+        materialInstance_->setParameter(parameter, filament::RgbaType::LINEAR,
+                                        color);
+    }
+
+    return *this;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetTexture(
+        const char* parameter,
+        const TextureHandle& textureHandle,
+        const TextureSamplerParameters& samplerConfig) {
+    if (materialInstance_) {
+        auto wTexture =
+                EngineInstance::GetResourceManager().GetTexture(textureHandle);
+
+        if (auto texturePtr = wTexture.lock()) {
+            materialInstance_->setParameter(
+                    parameter, texturePtr.get(),
+                    SamplerFromSamplerParameters(samplerConfig));
+        } else {
+            utility::LogWarning(
+                    "Failed to set texture for material.\n\tMaterial handle: "
+                    "{}\n\tTexture handle: {}\n\tParameter name: {}",
+                    currentHandle_, textureHandle, parameter);
+        }
+    }
+
+    return *this;
+}
+
+MaterialModifier& FilamentMaterialModifier::SetDoubleSided(bool doubleSided) {
+    if (materialInstance_) {
+        materialInstance_->setDoubleSided(doubleSided);
+    }
```

Comment:
I don't think we need to abort just because something went wrong somewhere. As a library, I don't really think we have to right to abort, actually.  Assert, sure. (Technically LogError throws an exception, but since none of the Open3D functions publicize that they might throw an exception if something goes wrong in the bowels, it's an abort for practical purposes, since the exception probably can't be caught anywhere recoverable.)